mmgen/core/evaluation/metrics.py

```diff
@@ -16,12 +17,110 @@
 from mmgen.models.architectures import InceptionV3
 from mmgen.models.architectures.common import get_module_device
 from mmgen.models.architectures.lpips import PerceptualLoss
+from mmgen.utils.io_utils import download_from_url
 from ..registry import METRICS
 from .metric_utils import (_f_special_gauss, _hox_downsample,
                            compute_pr_distances, finalize_descriptors,
                            get_descriptors_for_minibatch, get_gaussian_kernel,
                            laplacian_pyramid, slerp)
 
+TERO_INCEPTION_URL = 'https://nvlabs-fi-cdn.nvidia.com/stylegan2-ada-pytorch/pretrained/metrics/inception-2015-12-05.pt'  # noqa
+
+
+def load_inception(inception_args, metric):
+    """Load Inception Model from given ``inception_args`` and ``metric``. This
+    function would try to load Inception under the guidance of 'type' given in
+    `inception_args`, if not given, we would try best to load Tero's ones. In
+    detailly, we would first try to load the model from disk with the given
+    'inception_path', and then try to download the checkpoint from
+    'inception_url'. If both method are failed, pytorch version of Inception
+    would be loaded.
+
+    Args:
+        inception_args (dict): Keyword args for inception net.
+        metric (string): Metric to use the Inception. This argument would
+            influence the pytorch's Inception loading.
+    Returns:
+        model (torch.nn.Module): Loaded Inception model.
+        style (string): The version of the loaded Inception.
+    """
+
+    if not isinstance(inception_args, dict):
+        raise TypeError('eceive invalid \'inception_args\': '
+                        f'\'{inception_args}\'')
+
+    _inception_args = deepcopy(inception_args)
+    inceptoin_type = _inception_args.pop('type', None)
+
+    # load pytorch version
+    if inceptoin_type != 'StyleGAN':
+        return _load_inception_torch(_inception_args, metric), 'pytorch'
+
+    # try to load Tero's version
+    path = _inception_args.get('inception_path', None)
+    url = _inception_args.get('inception_url', TERO_INCEPTION_URL)
+
+    model = _load_inception_from_path(path)
+    if isinstance(model, torch.nn.Module):
+        return model, 'StyleGAN'
+
+    model = _load_inception_from_url(url)
```

Comment:
I just think the priority of `load_from_url` should be higher than `load_from_path`. Because we will automatically use the cached checkpoint once the file has been downloaded. The behavior we hope is that users can only provide the URL. Is it right?